src/ImageSharp/Formats/Tiff/TiffFrameMetadataExtensions.cs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using SixLabors.ImageSharp.Metadata.Profiles.Exif;
+
+namespace SixLabors.ImageSharp.Formats.Experimental.Tiff
+{
+    /// <summary>
+    /// The tiff metadata extensions
+    /// </summary>
+    internal static class TiffFrameMetadataExtensions
+    {
+        public static T[] GetArray<T>(this TiffFrameMetadata meta, ExifTag tag, bool optional = false)
+            where T : struct
+        {
+            if (meta.TryGetArray(tag, out T[] result))
+            {
+                return result;
+            }
+
+            if (!optional)
+            {
+                TiffThrowHelper.ThrowTagNotFound(nameof(tag));
+            }
+
+            return null;
+        }
+
+        public static bool TryGetArray<T>(this TiffFrameMetadata meta, ExifTag tag, out T[] result)
+            where T : struct
+        {
+            foreach (IExifValue entry in meta.FrameTags)
+            {
+                if (entry.Tag == tag)
+                {
+                    DebugGuard.IsTrue(entry.IsArray, "Expected array entry");
+
+                    result = (T[])entry.GetValue();
+                    return true;
+                }
+            }
+
+            result = null;
+            return false;
+        }
+
+        public static TEnum[] GetEnumArray<TEnum, TTagValue>(this TiffFrameMetadata meta, ExifTag tag, bool optional = false)
+          where TEnum : struct
+          where TTagValue : struct
+        {
+            if (meta.TryGetArray(tag, out TTagValue[] result))
+            {
+                return System.Array.ConvertAll(result, a => (TEnum)(object)a);
```

Comment:
unfortunately we cannot use `ConvertAll` here since `netstandard 1.3` does not implement this